### ISSUE TITLE
Wrong warning about linker input unused

### ISSUE THREAD

**Issue by Fedr** [llvm/llvm-project#56778]
Building this program
```c++
#include <spdlog/spdlog.h>
#include <spdlog/fmt/ostr.h>
#include <thread>

int main() {
    spdlog::info( "{}", std::this_thread::get_id() );
}
```
in Clang 14.0.0 with command line options `-std=c++20 -Wall -Wextra -pedantic-errors -lpthread` produces the warning:
```console
clang-14: warning: -lpthread: 'linker' input unused [-Wunused-command-line-argument]
```

But if one removes `-lpthread` option, then the build fails with the errors:
```console
/opt/compiler-explorer/gcc-11.2.0/lib/gcc/x86_64-linux-gnu/11.2.0/../../../../x86_64-linux-gnu/bin/ld: ./libspdlogd.a(async.cpp.o): in function `std::thread::thread<spdlog::details::periodic_worker::periodic_worker(std::function<void ()> const&, std::chrono::duration<long, std::ratio<1l, 1l> >)::$_0, , void>(spdlog::details::periodic_worker::periodic_worker(std::function<void ()> const&, std::chrono::duration<long, std::ratio<1l, 1l> >)::$_0&&)':
async.cpp:(.text+0xdc1): undefined reference to `pthread_create'
/opt/compiler-explorer/gcc-11.2.0/lib/gcc/x86_64-linux-gnu/11.2.0/../../../../x86_64-linux-gnu/bin/ld: ./libspdlogd.a(async.cpp.o): in function `std::thread::thread<spdlog::details::thread_pool::thread_pool(unsigned long, unsigned long, std::function<void ()>)::$_1, , void>(spdlog::details::thread_pool::thread_pool(unsigned long, unsigned long, std::function<void ()>)::$_1&&)':
async.cpp:(.text+0x2191): undefined reference to `pthread_create'
```
So the option is not "unused".

Online demo: https://gcc.godbolt.org/z/qxoWe8fzd

**Comment by DimitryAndric** [comment]
On Linux, you should generally use `-pthread`, not `-lpthread`. I guess this is why the driver warns, but it should probably suggest using `-pthread` instead, as a special case?

**Comment by Fedr** [comment]
Thanks, `-pthread` option removes the warning. A suggestion from clang would be nice!

**Comment by dwblaikie** [comment]
This isn't a clang issue, it's a godbolt issue, I think? it uses the same flags to do a separate assembly run (which doesn't link, so produces the linker warning) and then another compiler invocation that compiles+links (& does use/need the flag):
https://gcc.godbolt.org/z/qYPrf3TYG

**Comment by jiapei-nexera** [comment]
@dwblaikie 
Seems to me, this is the issue...

**Comment by AdityaDhotarkar** [comment]
@dwblaikie 
How can we run clang plugin on windows?
Cmake files generates .dll file but when it runs, shows blank.
